mods/cnc/maps/cnc64gdi01/cnc64gdi01.lua

```diff
@@ -0,0 +1,209 @@
+	
+	AIRepairBuildings(enemy)
+	AIRebuildHarvesters(enemy)
+
+	Utils.Do(AttackTriggers, function(a)
+		Trigger.OnKilledOrCaptured(a, function()
+			NodVehicleProduction(Utils.Random(AutocreateSquads))
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(150), AutoCreateTeam)
+	Trigger.AfterDelay(DateTime.Minutes(5), HeliHunt)
+
+	NodInfantryProduction()
+
+	Camera.Position = UnitsRally.CenterPosition
+	ObeliskFlare = Actor.Create('flare', true, { Owner = player, Location = Flare.Location })
+	Reinforce(CommandoReinforcements)
+end
+
+Tick = function()
+	if DateTime.GameTime > DateTime.Seconds(5) and player.HasNoRequiredUnits() then
+		player.MarkFailedObjective(destroyBiotechCenterObjective)		
+	end
+end
+
+Reinforce = function(units)
+	Media.PlaySpeechNotification(player, "Reinforce")
+	ReinforceWithLandingCraft(units, lstStart.Location, lstEnd.Location, UnitsRally.Location)
+end
+
+ReinforceWithLandingCraft = function(units, transportStart, transportUnload, rallypoint)
+	local transport = Actor.Create("oldlst", true, { Owner = player, Facing = 0, Location = transportStart })
+	local subcell = 0
+	Utils.Do(units, function(a)
+		transport.LoadPassenger(Actor.Create(a, false, { Owner = transport.Owner, Facing = transport.Facing, Location = transportUnload, SubCell = subcell }))
+		subcell = subcell + 1
+	end)
+
+	transport.ScriptedMove(transportUnload)
+
+	transport.CallFunc(function()
+		Utils.Do(units, function()
+			local a = transport.UnloadPassenger()
+			a.IsInWorld = true
+			a.MoveIntoWorld(transport.Location - CVec.New(0, 1))
+
+			if rallypoint ~= nil then
+				a.Move(rallypoint)
+			end
+		end)
+	end)
+
+	transport.Wait(5)
+	transport.ScriptedMove(transportStart)
+	transport.Destroy()
+end
+
+HuntTriggerFunction = function()
+	local list = enemy.GetGroundAttackers()
+	Utils.Do(list, function(unit)
+		IdleHunt(unit)
+	end)
+end
+
+IdleHunt = function(unit)
+	if not unit.IsDead then
+		Trigger.OnIdle(unit, function()
+			unit.AttackMove(UnitsRally.Location, 50)
+			unit.Hunt()
+		end)
+	end
+end
+
+NodInfantryProduction = function()
+	if HandOfNod.IsDead or HandOfNod.Owner == player then
+		return
+	end
+	HandOfNod.Build(inf1, SquadHunt)
+	Trigger.AfterDelay(DateTime.Seconds(15), NodInfantryProduction)
+end
+
+NodVehicleProduction = function(Squad)
+	if Airfield.IsDead or not Airfield.Owner == enemy then
+		return
+	end
+	Airfield.Build(Squad, SquadHunt)
+end
+
+AIRepairBuildings = function(ai)
+	Utils.Do(Map.NamedActors, function(actor)
+		if actor.Owner == ai and actor.HasProperty("StartBuildingRepairs") then
+			Trigger.OnDamaged(actor, function(building)
+				if building.Owner == ai and building.Health < 0.9 * building.MaxHealth then
+					building.StartBuildingRepairs()
+				end
+			end)
+		end
+	end)
+end
+
+HeliHunt = function()
+	local helicopters = enemy.GetActorsByType("heli")
+	SquadHunt(helicopters)
+end
+
+SquadHunt = function(actors)
+	Utils.Do(actors, function(actor)
+		Trigger.OnIdle(actor, function()
```

Comment:
This will need an `IsDead` check as well. It could happen that the first unit of a squad has already been killed while the last one is still being produced. As soon as it is finished, this would crash.